crates/rpc/src/method/get_storage_proof.rs

```diff
@@ -460,31 +462,53 @@ fn get_contract_storage_proofs(
         Some(contracts_storage_keys) => {
             let mut proofs = vec![];
             for csk in contracts_storage_keys {
-                let root = tx
-                    .contract_root_index(block_number, csk.contract_address)
-                    .context("Querying contract root index")?;
-
-                if let Some(root) = root {
-                    let nodes: Vec<NodeHashToNodeMapping> = ContractsStorageTree::get_proofs(
-                        &tx,
-                        csk.contract_address,
-                        block_number,
-                        &csk.storage_keys,
-                        root,
-                    )?
-                    .into_iter()
-                    .flatten()
-                    .map(|(node, node_hash)| NodeHashToNodeMapping {
+                if csk.storage_keys.is_empty() {
+                    let contract_root = tx
+                        .contract_root(block_number, csk.contract_address)
+                        .context("Querying contract's root")?
+                        .unwrap_or_default();
+
+                    let ContractRoot(node_hash) = contract_root;
+
+                    let merkle = TrieNode::Binary {
+                        left: Felt::default(),
+                        right: Felt::default(),
+                    };
+
+                    let node_map = NodeHashToNodeMapping {
                         node_hash,
-                        node: ProofNode(node),
-                    })
-                    .collect::<HashSet<_>>()
-                    .into_iter()
-                    .collect();
+                        node: ProofNode(merkle),
+                    };
 
-                    proofs.push(NodeHashToNodeMappings(nodes));
+                    proofs.push(NodeHashToNodeMappings(vec![node_map]));
```

Comment:
I think this is incorrect. A root of the storage trie for a particular contract is not a proof. As per my main comment - we should extend the API so that when you query for a proof of a contract you get it's `class`, `nonce` and `storage_root` in return. So the fix would be in `get_contract_proofs` instead of `get_contract_storage_proofs`.